### PR TITLE
Added an enumerated design space

### PR DESCRIPTION
**Pull request by latture**
This PR adds an enumerated design space defined by a list of python dictionaries. Users define a list of descriptors and a list of dictionaries that map from descriptor key to value. Values are allowed to be any type, i.e. `data: Mapping[str, Any]`. The backend casts the value to a string when building the design space.

### CODE REVIEW

**Comment by latture** [comment]
Rebuild me!

**Comment by latture** [comment]
Should we add an integration test that ensures the enumerated processor and design space can be used in a workflow?

File: tests/serialization/test_design_spaces.py [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
-from citrine.informatics.design_spaces import DesignSpace, ProductDesignSpace
+from citrine.informatics.descriptors import CategoricalDescriptor, RealDescriptor
+from citrine.informatics.design_spaces import DesignSpace, ProductDesignSpace, EnumeratedDesignSpace
 from citrine.informatics.dimensions import ContinuousDimension, EnumeratedDimension
 
 
 @pytest.fixture
-def valid_data():
-    """Produce valid design space data."""
+def valid_product_data():
```
Comment (andrewmillspaugh) [inline tests/serialization/test_design_spaces.py]: Might be a good place to check out factory-boy. See https://github.com/CitrineInformatics/citrine-python/blob/develop/tests/utils/factories.py

**Comment by latture** [comment]
Tests fail due to insufficient test coverage:
```
FAIL Required test coverage of 91% not reached. Total coverage: 90.57%
```

File: tests/serialization/test_design_spaces.py (hunk 1 above)
Comment (latture) [reply tests/serialization/test_design_spaces.py]: That's a good idea. If we don't end up auto-generating these classes we should refactor the orion tests. There's quite a bit of code duplication.

**Comment by latture** [comment]
I added a few tests in #40 that bump up test coverage. The build is passing now.

**Pull request merged**